### DISCUSSION TITLE
Trying to run test notification generates error on Unraid

### DISCUSSION THREAD

**Discussion by Zxurian** [jef/streetmerchant#1419] [Help]
I have the docker container installed on Unraid, and everything looks like it's setup and running correctly, however when I try to run a test notification, I get an error. The only notification i'm using is DISCORD_WEBHOOK
```
/app $ npm run test:notification

> streetmerchant@ test:notification /app
> npm run build && node build/__test__/notification-test.js


> build
> tsc

sh: tsc: not found
npm notice 
npm notice New patch version of npm available! 7.1.1 -> 7.1.2
npm notice Changelog: https://github.com/npm/cli/releases/tag/v7.1.2
npm notice Run npm install -g npm@7.1.2 to update!
npm notice 
npm ERR! code 127
npm ERR! path /app
npm ERR! command failed
npm ERR! command sh -c tsc

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/appuser/.npm/_logs/2020-12-15T04_42_28_575Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! streetmerchant@ test:notification: `npm run build && node build/__test__/notification-test.js`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the streetmerchant@ test:notification script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/appuser/.npm/_logs/2020-12-15T04_42_28_791Z-debug.log
/app $ 
```

any ideas on how I can test the notifications to make sure they're working?

**Comment by ShokoPeak** [comment]
Could we take a look at your dotenv configuration?

Also, you're generating a couple of different error files, have you tried checking back through them for more information?

**Reply by Zxurian** [reply]
so I don't have a specific dotenv file. With Unraid, I'm setting port/path variables up through the Unraid UI. The following is every key/value pair I've set on the container. I also haven't checked through the individual files no. I went to attach them here, and it says that `/home/appuser/.npm` isn't a valid directory on the container.

![image](https://user-images.githubusercontent.com/573371/102507187-f57bec00-4051-11eb-9d11-0daa85f2a87b.png)
![image](https://user-images.githubusercontent.com/573371/102507243-062c6200-4052-11eb-9009-5ced84f3aa80.png)

I could also be trying to initiate a test wrong, since I'm not 100% familiar with docker. The `npm run test:notification` command i'm running from inside the docker container itself.

**Reply by ShokoPeak** [reply]
I've had similar issues with docker in the past for other stuff, could there be a permissions error maybe that's stopping docker recognising the `/home/appuser/.npm` as valid? All your other config looks OK to me, but it might be more of a docker troubleshooting issue maybe. Sorry, i'm not great with Docker myself but I would definitely look into it being a permissions issue.

**Comment by dthomas710** [comment]
Fancy seeing you here again :) 

Just wanted to throw my hat into the ring as I'm having the exact same issue

![image](https://user-images.githubusercontent.com/68824903/102523690-be570c00-404c-11eb-949b-5ca570815f98.png)

EDIT: 

Adding a pathway for /home does NOT fix the issue and adding a pathway for /app breaks the container

**Reply by dthomas710** [reply]
Ok well with a bit of anecdotal evidence I can confirm notifications are working on my end even if the testing isn't, and that is because the bot just WORKED TO NOTIFY ME ABOUT A WALMART DROP AND I GOT A CONSOLE WOOOOOOOOO!!!!!!

![image](https://user-images.githubusercontent.com/68824903/102542864-4813d300-4067-11eb-99b9-cc7ab0723b7e.png)

![image](https://user-images.githubusercontent.com/68824903/102542744-169b0780-4067-11eb-9495-db3070e38d45.png)

@Zxurian I want to say that you're probably in the same boat as me since our setups are very similar, like I would assume the notis are still working despite the failed tests, but no way to know for sure unfortunately. Thank you SO MUCH again for your help, seriously couldn't have done this without you. I really hope you get your 3080 ASAP. 

@jef you're an amazing human. Thank you for creating this stellar tool.

**Reply by jef** [reply]
Congrats @dthomas710!

**Comment by jef** [comment]
I'm sorry you've been cycling on this... You're missing one little thing. When you start your container, you'll want to test with `test:notification:production`. I'm not 100% on how you're starting your containers in Unraid, but you'll want to either set `CMD` to `test:notification:production` or call it within the container. Sorry if that's confusing, it's just that I haven't ran inside Unraid before.

**Reply by Zxurian** [reply]
it's no problem. I saw the info on starting the container with the extra information, and that worked for me. (incidentally if someone is using Unraid, then that string would be added as a Post Argument under Advanced View)
![image](https://user-images.githubusercontent.com/573371/102631297-88cd2400-411b-11eb-8ff4-bc9ae33c8dfe.png)

This was more trying to get the runtime test to work as shown here: https://jef.codes/streetmerchant/reference/notification/ It might just be how to initiate the test from an Unraid hosted container. I thought that logging in to the container directly and running that would send out a test notification.
